src/bin/lpython.cpp

```diff
@@ -886,6 +886,8 @@ int interactive_python_repl(
             res = fe.evaluate(code_string, verbose, lm, pass_manager, diagnostics);
             if (res.ok) {
                 r = res.result;
+                std::cerr << diagnostics.render(lm, compiler_options);
+                diagnostics.clear();
```

Comment:
On second thought, I think this approach is good to print style info/warnings. But for printing errors (semantic errors like in https://github.com/lcompilers/lpython/issues/2757), we should fix the `python_ast_to_asr` to return the result as not ok (that is `res.ok = False`).

Reply:
I created a new issue for this here https://github.com/lcompilers/lpython/issues/2810. I think this PR is good to merge.

Reply:
I noticed this as well. The reason is that for the intrinsic functions, we used `append_error` instead of throwing a semantic error. This could potentially be resolved by throwing the error later from `python_ast_to_asr` if the intrinsic function returns null.